### ISSUE TITLE
Speed of LSTM

### ISSUE THREAD

**Issue by edf42001** [davidbernat/work-summer-2019#3]
I was working on a simple LSTM network, and, as expected, it is training extremely slowly. I was wondering if your cloud based networks will be faster than my computer?

**Comment by davidbernat** [comment]
No, it won't be. Though, what are you training and what are the parameters
of the network?

**Comment by edf42001** [comment]
I was following this tutorial for generating a short story: https://towardsdatascience.com/lstm-by-example-using-tensorflow-feb0c1968537. I'm moving on to time series now. 

 The network had two LSTM cells with 512 hidden units each. How that translates to the forget, remember, and whatever other gates are inside an LSTM cell, I do not know, although I'm looking it up right now. Learning rate is 0.001 using an RMSProp optimizer. The tutorial suggested training for 50,000 iterations, which would take ~25 minutes on my computer. Based on what I read, this is reasonable and I just need more patience.

**Comment by davidbernat** [comment]
See the attached for a 1 sensor, 1 variable simulation of data. It is a
noisy sinusoid that also periodically clicks (i.e., a sharp, very short
spike). To make it more realistic, the intensity (not frequency) of the
clicks increase over time. Let me know if you have any questions.

A few things:
1. Use AdaProp
2. 512 is a lot of hidden units. Find somewhere a heuristic for how many to
use. Generally it should be about the square root of the number of
dimensions of your features (if I remember correctly). 16 or 32 is usually
a safe bet. In the data attached there is only 1 feature.
3. I don't see how much data the tutorial uses. But the model for the
example here should be tiny and you should see that it trains very fast.

Let me know if you have any other questions.

**Comment by edf42001** [comment]
Sorry, I can't find an attachment. And I was having trouble finding anything for AdaProp, but eventually I found that it is an optimizer. I'm not sure it's directly included in tensorflow, so I'll have to look more.

**Comment by davidbernat** [comment]
I'm going to check a few files into the repository.

**Comment by davidbernat** [comment]
Quick question as a quiz: when doing sequence modeling for language, what
is the number of features involved? It's one string of text, yes, but how
many features?
